hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java

```diff
@@ -2841,6 +2845,9 @@ public MultiResponse multi(final RpcController rpcc, final MultiRequest request)
             spaceQuotaEnforcement);
         }
       } finally {
+        if (context != null) {
```

Comment:
I dont think this works out so well, because of how some of the methods are abstracted. For example, `increment` can be called from the single `mutate` method and also from `doNonAtomicRegionMutation` (which is called from `multi`). So for increments, we'd double account them.

I might recommend only adding calls to addBlockBytesScanned wherever there are existing calls to getBlockBytesScanned() for metrics. As I mentioned in another comment, you'll want to move the getBlockBytesScanned() outside the if (metrics != null) checks, but otherwise those are probably the best places to add the quota calls if possible.

Reply:
Hmm actually maybe it's better to keep this here and to instead handle avoiding double counting append/increment, i.e. by passing in a isMulti boolean or something